cpp/src/arrow/csv/writer.cc

```diff
@@ -146,7 +165,7 @@ class ColumnPopulator {
 
  protected:
   virtual Status UpdateRowLengths(int64_t* row_lengths) = 0;
-  std::shared_ptr<StringArray> casted_array_;
+  std::shared_ptr<Array> array_;
```

Comment:
Can you just a comment for `array_` that it would be a `StringArray` or `LargeStringArray`?

Reply:
Fixed.